### PR TITLE
Usher remove factories

### PR DESCRIPTION
**Pull request by jbachhardie**
Moving usher to a simpler, not factory-based structured, with a YAML syntax based on Ansible.

https://trello.com/c/b8AItTjR/241-usher

### CODE REVIEW

**Comment by bapti** [comment]
:+1:

File: src/run.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const logger = require('winston');
+const parser = require('js-yaml');
+const fs = require('fs');
+const _ = require('lodash');
+
+let spawnSync = require('child_process').spawnSync;
+
+function expandTokens(command, vars) {
```
Comment (bapti) [inline src/run.js:10]: @jbachhardie @AtebMT 

How do we feel here about functions? From what I've seen in OS land folk seem to prefer this style

``` js
const expandTokens = () => {}
```

File: src/run.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  logger.info(`Executing command : ${command.join(" ")}`);
+
+  return spawnSync(command[0], _.tail(command), spawnOptions)
+}
+
+module.exports = (taskName, taskVars, opts) => {
+  const usherFile = opts.filepath || '.usher.yml';
+  const parsedVars = resolveKeyValuePairs(taskVars);
+  const config = parser.safeLoad(fs.readFileSync(usherFile, 'utf8'));
+  const vars = _.merge(config.vars, parsedVars);
```
Comment (bapti) [inline src/run.js]: Feels like we could merge these 4 lines into an extracted function - `buildConfig` or something like that

Comment (AtebMT) [inline src/run.js]: Done!

File: src/run.js (hunk 1 above)
Comment (AtebMT) [inline src/run.js:10]: Personally, I like that function is an obvious key. => {} is typically used with anonymous functions, so why give it a name? Also, it changes the value of the this object, which might confuse some people who aren't aware that "this" in a function is different from "this" in a lambda.

**Pull request merged**